src/payment-card/payment-card.js

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright (c) 2018-2019 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import valid from 'card-validator';
+
+import {getOverrides} from '../helpers/overrides.js';
+import {Input, SIZE} from '../input/index.js';
+import {ThemeContext} from '../styles/theme-provider.js';
+
+import AmexIcon from './icons/amex.js';
+import DinersClubIcon from './icons/dinersclub.js';
+import DiscoverIcon from './icons/discover.js';
+import EloIcon from './icons/elo.js';
+import GenericIcon from './icons/generic.js';
+import JcbIcon from './icons/jcb.js';
+import MaestroIcon from './icons/maestro.js';
+import MastercardIcon from './icons/mastercard.js';
+import UnionPayIcon from './icons/unionpay.js';
+import VisaIcon from './icons/visa.js';
+
+import {IconWrapper as StyledIconWrapper} from './styled-components.js';
+
+import type {PaymentCardPropsT} from './types.js';
+
+const CardTypeToComponent = {
+  visa: VisaIcon,
+  mastercard: MastercardIcon,
+  'american-express': AmexIcon,
+  'diners-club': DinersClubIcon,
+  discover: DiscoverIcon,
+  jcb: JcbIcon,
+  unionpay: UnionPayIcon,
+  maestro: MaestroIcon,
+  elo: EloIcon,
+  generic: GenericIcon,
+};
+
+export const addGaps = (gaps: number[], value: string) =>
+  gaps.reduce(
+    (prev, gap, index) =>
+      `${prev.slice(0, gap + index)} ${prev.slice(gap + index)}`.trim(),
+    `${value}`,
+  );
+
+export const sanitizeNumber = (input: string) => {
+  const number = input.replace(/[^0-9]/gi, '');
+  const validatedValue = valid.number(number);
+  if (validatedValue.card && Array.isArray(validatedValue.card.lengths)) {
+    return number.slice(
+      0,
+      validatedValue.card.lengths[validatedValue.card.lengths.length - 1],
+    );
+  }
+  // CC number NEVER can have more than 19 digits
+  return number.slice(0, 19);
```

Comment:
can we limit the length based on the credit card type?

Reply:
Yep, that's what happens right above! This is a fallback when the type is not detected (yet).

Reply:
oh, I've overlooked it, great!